extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -715,6 +715,26 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return {'artc': artc.join() || null, 'ati': ati.join(), 'ard': ard.join()};
   }
 
+  /** @override */
+  getCustomRealTimeConfigMacros_() {
+    return {
+      PAGEVIEWID: () => {
+        const docInfo = Services.documentInfoForDoc(this.element);
+        return docInfo.pageViewId;
+      },
+      HREF: () => this.win.location.href,
+      DATASLOT: () => this.element.getAttribute('data-slot'),
+      HEIGHT: () => this.element.getAttribute('height'),
+      WIDTH: () => this.element.getAttribute('width'),
+      MULTISIZE: () => this.element.getAttribute('data-multi-size'),
+      MULTISIZE_VALIDATION:
+      () => this.element.getAttribute('data-multi-size-validation'),
+      OVERRIDEWIDTH: () => this.element.getAttribute('data-override-width'),
+      OVERRIDEHEIGHT: () => this.element.getAttribute('data-override-height'),
+      JSON: () => this.element.getAttribute('data-json'),
```

Comment:
This could greatly increase the length of the url.   Do we truncate the url if its too long?  Do we indicate it was truncated?  Is there only a portion of the JSON that they want?

Reply:
added truncation to 16k